### ISSUE TITLE
How to display sheet after hiding

### ISSUE THREAD

**Issue by CripyIce** [gordontucker/FittedSheets#122]
Hi there,
I'm displaying a `SheetViewController` using inline method, and after calling `sheetController?.animateOut()` the method `sheetController?.animateIn()` does nothing.
I could only redisplay the `SheetViewController` after calling all this code again:
```swift
sheetController.willMove(toParent: self)
self.addChild(sheetController)
view.addSubview(sheetController.view)
sheetController.didMove(toParent: self)

sheetController.view.translatesAutoresizingMaskIntoConstraints = false
NSLayoutConstraint.activate([
    sheetController.view.topAnchor.constraint(equalTo: view.topAnchor),
    sheetController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor),
    sheetController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
    sheetController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor)
])

// animate in
sheetController.animateIn()
```

Am I using this library the wrong way? Or is it how it's supposed to be?
Thanks!

**Comment by danishnisar** [comment]
Facing the same issue.

**Comment by gordontucker** [comment]
This should be fixed in 2.3.0

**Comment by CripyIce** [comment]
@gordontucker This is still happens...

**Comment by gordontucker** [comment]
Have you looked at the demo in the repo? It has an inline example that animates out and in multiple times (Use the `Recycled Sheet` for it to do that). Does that work for you? That is the test I added to make sure reusing the sheet would work as expected and I did find a bug with it that I fixed in 2.3.0.

**Comment by ataravati** [comment]
@gordontucker I'm having the same issue. Your demos are very hard to follow to be honest. Also, regardless of that, this is a bug. `animateOut()` should just animate the sheet out of the screen; exactly the opposite of `animateIn()` which just animates the sheet into the screen. If all the following is not done as part of the `animateIn()`, you shouldn't have to repeat them after calling `animateOut()`.

```
sheetController.willMove(toParent: self)
self.addChild(sheetController)
view.addSubview(sheetController.view)
sheetController.didMove(toParent: self)

sheetController.view.translatesAutoresizingMaskIntoConstraints = false
NSLayoutConstraint.activate([
    sheetController.view.topAnchor.constraint(equalTo: view.topAnchor),
    sheetController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor),
    sheetController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
    sheetController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor)
])
```

**Comment by gordontucker** [comment]
I added another update that might help you out. The call to animate in should change from this:

```sheetController.willMove(toParent: self)
self.addChild(sheetController)
view.addSubview(sheetController.view)
sheetController.didMove(toParent: self)

sheetController.view.translatesAutoresizingMaskIntoConstraints = false
NSLayoutConstraint.activate([
    sheetController.view.topAnchor.constraint(equalTo: view.topAnchor),
    sheetController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor),
    sheetController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
    sheetController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor)
])

// animate in
sheetController.animateIn()```

to this:

```
sheetController.animateIn(to: view, in: self)
```

Once you switch to use that and have 2.3.2, it should fix the problem you are having. Thanks for being patient and not being complaining about a free tool not being good enough. I appreciate your patiences everyone.

**Comment by fridayhoney** [comment]
still happens...

**Comment by fridayhoney** [comment]
It works but the sheetController appear inside the navigationviewcontroller...

**Comment by gordontucker** [comment]
Did you add it inside the navigation view controller? If so, that is how it is expected to work.
